docs/tools/terraform/howto/vnet-peering-azure.rst

```diff
@@ -0,0 +1,214 @@
+
+      permissions {
+        actions = ["Microsoft.Network/virtualNetworks/peer/action"]
+      }
+
+      assignable_scopes = [
+        "/subscriptions/${data.azurerm_subscription.subscription.subscription_id}"
+      ]
+    }
+
+    # 9. Assign the custom role to the Aiven service principal
+    resource "azurerm_role_assignment" "aiven_role_assignment" {
+      role_definition_id = azurerm_role_definition.role_definition.role_definition_resource_id
+      principal_id       = azuread_service_principal.aiven_app_principal.object_id
+      scope              = azurerm_virtual_network.virtual_network.id
+
+      depends_on = [
+        azuread_service_principal.aiven_app_principal,
+        azurerm_role_assignment.app_role
+      ]
+    }
+
+    # 10. Find your AD tenant id
+    # Skip, it's in the env
+
+    # 11. Create a peering connection from the Aiven Project VPC
+    # 12. Wait for the Aiven platform to set up the connection
+    resource "aiven_azure_vpc_peering_connection" "peering_connection" {
+      vpc_id                = aiven_project_vpc.avn_vpc.id
+      peer_resource_group   = azurerm_resource_group.resource_group.name
+      azure_subscription_id = data.azurerm_subscription.subscription.subscription_id
+      vnet_name             = azurerm_virtual_network.virtual_network.name
+      peer_azure_app_id     = azuread_application.app.application_id
+      peer_azure_tenant_id  = "00000000-0000-0000-0000-000000000000"
+
+      depends_on = [
+        azurerm_role_assignment.aiven_role_assignment
+      ]
+    }
+
+
+Create peering in Azure
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Now we need to create connection using credentials from the previous step.
+Unfortunately terraform doesn't support `dynamic provider configuration <https://github.com/hashicorp/terraform/issues/25244>`_.
+
+.. code-block::
+
+    # 13. Create peering from your VNet to the Project VPC's VNet
+    provider "azurerm" {
+      features {}
+      alias                = "app"
+      client_id            = azuread_application.app.application_id
+      client_secret        = azuread_application_password.app_password.value
+      subscription_id      = data.azurerm_subscription.subscription.subscription_id
+      tenant_id            = "00000000-0000-0000-0000-000000000000"
+      auxiliary_tenant_ids = [azuread_service_principal.aiven_app_principal.application_tenant_id]
+    }
+
+    resource "azurerm_virtual_network_peering" "network_peering" {
+      provider                     = azurerm.app
+      name                         = "my-azure-virtual-network-peering"
+      remote_virtual_network_id    = aiven_azure_vpc_peering_connection.peering_connection.state_info["to-network-id"]
+      resource_group_name          = azurerm_resource_group.resource_group.name
+      virtual_network_name         = azurerm_virtual_network.virtual_network.name
+      allow_virtual_network_access = true
+    }
+
+    # 14. Wait until the Aiven peering connection is active
+
+Done.
```

Comment:
Is there any way users can check whether the connection is active?

Reply:
Well, terraform can't tell you things "are ready". You can check internal state for that by running terraform again and again and reading state file until you get desired field got updated. 
But this is not very _right_ way. Unfortunately there is no convenient way to do that. I guess we can suggest go for cli for the step 14.
This is exactly what this article is not about: setup cli.  
That is a good question.

Reply:
That makes sense. Let's leave the CLI part out of this article. 👍